### PR TITLE
Py2 compatibility: add Numenta, NumentaTM, htmjava detectors

### PR DESCRIPTION
**Pull request by breznak**
- [x] allows to install (old) detectors that depend on python 2 to an automatic virtualenv
- [x] re-adds Numenta, NumentaTM, htmjava detectors
- [x] I think this has been running, probably made some errs in rebasing to this repo, as this is now failing
  - fixed by adding custom `run.py` lost during merge

TODO for a separate task: 
- [ ] make py2 install scripts platform independent (currently assumes UNIX) 
  - `install_py2_detectors.sh`

This has also been submitted to numenta/NAB https://github.com/iandanforth/NAB/pull/9

### CODE REVIEW

File: install_py2_detectors.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (breznak) [inline install_py2_detectors.sh:1]: this is currently a bash shell script, so will only run where bash is installed (typically Linux, UNIX), 
Future TODO: make multiplatform. Could this be a python script?

File: nab/detectors/numenta/setup.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    reqs = requirements
+  return reqs
+
+
+
+if __name__ == "__main__":
+  requirements = findRequirements()
+
+  setup(
+    name="nab-detector-numenta",
```
Comment (breznak) [inline nab/detectors/numenta/setup.py:100]: custom setup.py and run.py files for Numenta*, htmjava detectors

File: run.py [hunk 3]
```diff
@@ -37,9 +38,35 @@ def getDetectorClassConstructors(detectors):
...
+        argss.append("--dataDir="+str(args.dataDir))
+      if args.profilesFile:
+        argss.append("--profilesFile="+str(args.profilesFile))
+      if args.resultsDir:
+        argss.append("--resultsDir="+str(args.resultsDir))
+      if args.windowsFile:
+        argss.append("--windowsFile="+str(args.windowsFile))
+
+  if "htmjava" in py2detectors: # htm.java
+    subprocess.call([ENV_PY2, "./nab/detectors/htmjava/run.py"]+ argss)
```
Comment (breznak) [inline run.py:62]: special handling for python2 detectors that allows seamless integration from the main py3 caller.

File: run.py [hunk 4]
```diff
@@ -132,9 +159,9 @@ def main(args):
   parser.add_argument("-d", "--detectors",
                     nargs="*",
                     type=str,
-                    default=["null", "numenta", "random", "bayesChangePt",
-                             "windowedGaussian", "expose", "relativeEntropy",
-                             "earthgeckoSkyline"],
+                    default=["numenta", "numentaTM", "htmjava", "null", "random",
```
Comment (breznak) [inline run.py:162]: numenta, htmjava detectors are back!

**Review by breznak — COMMENTED** [review]
@ctrl-z-9000-times  please review, this is working for me

**Comment by breznak** [comment]
Please review when you have time, most of the large file changes is just copied from the py2 branch of NAB.

**Comment by D-McDougall** [comment]
Cool!
If I may ask, how did you test this?  What were the results?

**Comment by breznak** [comment]
> If I may ask, how did you test this? 

yes, locally. All combinations (py2,3) of detectors install and work seamlessly. That is the goal of this PR. 

> What were the results?

I don't quite understand. This is not (yet) our `htm.core`, "just" original numenta's Numenta & NumentaTM detectors using py2. 
- I do get similar, but slightly different results. But I do have the same behavior even with upstream numenta/NAB (due to random fluctuations) I guess the slight changes (with overall same score) should be ok? 
- our htm.core detector is queued and waiting for this to get merged ;)

**Review by D-McDougall — APPROVED** [review]
Ok, Thanks for improving this!

**Pull request merged**